src/app/clusters/scenes/SceneTableImpl.cpp

```diff
@@ -0,0 +1,680 @@
+        ReturnErrorOnFailure(mStorageData.mExtensionFieldSets.Deserialize(reader));
+
+        return reader.ExitContainer(container);
+    }
+};
+
+// A Full fabric serialized TLV length is 88 bytes, 128 bytes gives some slack tested bu running writer.GetLengthWritten at the
+// end of the Serialize method of FabricSceneData
+static constexpr size_t kPersistentFabricBufferMax = 128;
+
+/**
+ * @brief Linked list of all scenes in a fabric, stored in persistent memory
+ *
+ * FabricSceneData is an access to a linked list of scenes
+ */
+struct FabricSceneData : public PersistentData<kPersistentFabricBufferMax>
+{
+    FabricIndex fabric_index = kUndefinedFabricIndex;
+    uint8_t scene_count      = 0;
+    SceneStorageId scene_map[kMaxScenePerFabric];
+    FabricIndex next = kUndefinedFabricIndex;
+
+    FabricSceneData() = default;
+    FabricSceneData(FabricIndex fabric) : fabric_index(fabric) {}
+
+    CHIP_ERROR UpdateKey(StorageKeyName & key) override
+    {
+        VerifyOrReturnError(kUndefinedFabricIndex != fabric_index, CHIP_ERROR_INVALID_FABRIC_INDEX);
+        key = DefaultStorageKeyAllocator::FabricScenesKey(fabric_index);
+        return CHIP_NO_ERROR;
+    }
+
+    void Clear() override
+    {
+        scene_count = 0;
+        for (uint8_t i = 0; i < kMaxScenePerFabric; i++)
+        {
+            scene_map[i].Clear();
+        }
+        next = kUndefinedFabricIndex;
+    }
+
+    CHIP_ERROR Serialize(TLV::TLVWriter & writer) const override
+    {
+        TLV::TLVType container;
+        ReturnErrorOnFailure(writer.StartContainer(TLV::AnonymousTag(), TLV::kTLVType_Structure, container));
+
+        ReturnErrorOnFailure(writer.Put(TLV::ContextTag(kTagSceneCount), static_cast<uint16_t>(scene_count)));
+
+        // Storing the scene map
+        for (uint8_t i = 0; i < kMaxScenePerFabric; i++)
+        {
+            ReturnErrorOnFailure(writer.Put(TLV::ContextTag(kTagSceneEndpointID), static_cast<uint16_t>(scene_map[i].mEndpointId)));
+            ReturnErrorOnFailure(writer.Put(TLV::ContextTag(kTagSceneGroupID), static_cast<uint16_t>(scene_map[i].mGroupId)));
+            ReturnErrorOnFailure(writer.Put(TLV::ContextTag(kTagSceneID), static_cast<uint8_t>(scene_map[i].mSceneId)));
+        }
+        ReturnErrorOnFailure(writer.Put(TLV::ContextTag(kTagNext), static_cast<uint16_t>(next)));
+
+        return writer.EndContainer(container);
+    }
+
+    CHIP_ERROR Deserialize(TLV::TLVReader & reader) override
+    {
+        ReturnErrorOnFailure(reader.Next(TLV::AnonymousTag()));
+        VerifyOrReturnError(TLV::kTLVType_Structure == reader.GetType(), CHIP_ERROR_INTERNAL);
+
+        TLV::TLVType container;
+        ReturnErrorOnFailure(reader.EnterContainer(container));
+
+        ReturnErrorOnFailure(reader.Next(TLV::ContextTag(kTagSceneCount)));
+        ReturnErrorOnFailure(reader.Get(scene_count));
+        for (uint8_t i = 0; i < kMaxScenePerFabric; i++)
```

Comment:
So if `kMaxScenePerFabric` increases, this code will break, right?  We should either serialize how many scenes we had in there, or just keep looping until `Next()` returns END_OF_TLV when looking for kTagSceneEndpointID.